### ISSUE TITLE
losses and accuracies for training data only?

### ISSUE THREAD

**Issue by chaoshengdong** [Xi-L/ParetoMTL#1]
Hi, 

I think your paper is very interesting. While diving deep in your code, I have the following question.

Are the losses and accuracies reported in your code for training data only? Shall we report the testing loss and accuracy. 

Thank you!

**Comment by Xi-L** [comment]
Hi,

Glad to know you find our work is interesting!

In our paper, all the reported losses and accuracies are on the test set. 

In the code, we only report the training losses and accuracies during training. You can simply replace the train_loader with test_loader to obtain the testing performance, since the test loader is also already defined in the code.

Xi

**Comment by Weiren1996** [comment]
Hi,

I think your paper is very interesting. While diving deep in your code, I have the following question.

def circle_points(r, n):
    """
    generate evenly distributed unit preference vectors for two tasks
    """
    circles = []
    for r, n in zip(r, n):
        t = np.linspace(0, 0.5 * np.pi, n)
        x = r * np.cos(t)
        y = r * np.sin(t)
        circles.append(np.c_[x, y])
    return circles

The above method is used to generate uniformly distributed unit preference vectors for two tasks, but what if we are aiming for three tasks or even more? How to change it?

Thank you!
